src/rust-crypto/CrossSigningInfo.ts

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2023 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { CrossSigningInfo as ICrossSigningInfo } from "../crypto-api/CrossSigningInfo";
+import { CrossSigningKey } from "../crypto-api";
+import { IHttpOpts, MatrixHttpApi, Method } from "../http-api";
+import { IDownloadKeyResult, Keys } from "../client";
+
+/**
+ * Implementation of {@link ICrossSigningInfo}
+ */
+export class CrossSigningInfo implements ICrossSigningInfo {
+    /**
+     * Create an instance of {@link CrossSigningInfo} for the given user
+     *
+     * @param userId - the owner of the cross signing info
+     * @param http - http interface
+     */
+    public static async create(
+        userId: string,
+        http: MatrixHttpApi<IHttpOpts & { onlyData: true }>,
+    ): Promise<CrossSigningInfo | null> {
+        const queryBody = {
+            device_keys: {
+                [userId]: [],
+            },
+        };
+
+        const keyResult = await http.authedRequest<IDownloadKeyResult>(
+            Method.Post,
+            "/_matrix/client/v3/keys/query",
+            undefined,
+            queryBody,
+            {
+                prefix: "",
+            },
+        );
```

Comment:
given the method is called `get`**`Stored`**`CrossSigningForUser`, it seems odd that we should pull this from the server.

I think we need to call `OlmMachine::getIdentity`